### PR TITLE
SLVS-1397 Add ServerConnection model

### PR DESCRIPTION
**Pull request by georgii-borovinskikh-sonarsource**

### CODE REVIEW

File: src/Core.UnitTests/Binding/ServerConnectionTests.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+[TestClass]
+public class ServerConnectionTests
+{
+    [TestMethod]
+    public void Ctor_SonarCloudOrg_HasCorrectTypeAndDefaultSettings()
+    {
+        var serverConnection = new ServerConnection("Sonar Cloud Org");
+        
+        serverConnection.Type.Should().Be(ServerConnectionType.SonarCloud);
```
Comment (gabriela-trutan-sonarsource) [inline src/Core.UnitTests/Binding/ServerConnectionTests.cs:35]: Why are we not checking that the id was set? Same feedback for all the other tests

File: src/Core/Binding/ServerConnection.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        Id = id ?? throw new ArgumentNullException(nameof(id));
+        Type = type;
+        Settings = settings ?? throw new ArgumentNullException(nameof(settings));
+    }
+}
+
+public static class ServerConnectionExtensions
+{
+    public static Uri GetSonarQubeUri(this ServerConnection connection) =>
```
Comment (gabriela-trutan-sonarsource) [inline src/Core/Binding/ServerConnection.cs:63]: Why have these methods as extension methods and not as static methods? There is no gain. The serialization will simply ignore them.

**Review by gabriela-trutan-sonarsource — APPROVED** [review]

**Review by gabriela-trutan-sonarsource — COMMENTED** [review]
I do have some minor feedback

File: src/Core/Binding/ServerConnection.cs (hunk 2 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/Core/Binding/ServerConnection.cs:63]: You mean not as instance methods? I can move them there, sure

Comment (gabriela-trutan-sonarsource) [reply src/Core/Binding/ServerConnection.cs:63]: Yes, I meant instance methods.

**Review by gabriela-trutan-sonarsource — APPROVED** [review]

**Pull request merged**